mcs/class/corlib/System.Reflection.Emit/ModuleBuilder.cs

```diff
@@ -376,44 +369,33 @@ public override Type GetType (string className, bool throwOnError, bool ignoreCa
 			if (className.Length == 0)
 				throw new ArgumentException ("className");
 
-			int subt;
-			string orig = className;
-			string modifiers;
 			TypeBuilder result = null;
 
 			if (types == null && throwOnError)
 				throw new TypeLoadException (className);
 
-			subt = className.IndexOfAny (type_modifiers);
-			if (subt >= 0) {
-				modifiers = className.Substring (subt);
-				className = className.Substring (0, subt);
-			} else
-				modifiers = null;
+			TypeSpec ts = TypeSpec.Parse (className);
 
 			if (!ignoreCase) {
-				result =  name_cache [className] as TypeBuilder;
+				var displayNestedName = ts.WithoutModifiers().TypeName;
```

Comment:
Yea, there's not really a good payoff to `WithoutModifiers()`.  I thought I'd get to use it in a couple of different places, but there really aren't a lot of other APIs that need to manipulate type names in a very fancy way.  I'll make something simpler just for this one use.